shared/src/cred.rs

```diff
@@ -70,6 +70,158 @@ impl CredentialRPK {
     }
 }
 
+#[derive(Debug, Copy, Clone)]
+enum ByValueStyle {
+    /// The value of the credential is a CCS. It is encoded directly in CBOR.
+    KCCS,
+    // TBD: Add X509 and others, possibly also as ByteEncoded(key)
+}
+
+// From https://www.iana.org/assignments/cose/cose.xhtml#header-parameters
+const COSE_HEADER_PARAMETER_KCCS: u8 = 14;
+
+impl ByValueStyle {
+    fn encode(&self, value: &[u8]) {
+        todo!("and add a an encode buffer")
+    }
+
+    // and decode
+}
+
+/// A credential along with its corresponding public key
+///
+/// It may contain information about the key ID, which enables its use by reference, and of its
+/// value type, which enables its use by value.
+///
+/// Future extensions may also add other optional identifiers (eg. an identifying URI) that might
+/// enable additional kinds of referncing the credential; some of those may also not be stored but
+/// merely generated (eg. thumbprints, although in that case the type of credential needs to be
+/// stored, so that it can be decided whether the thumbprint is an x5t or a c5t).
+#[derive(Clone, Copy, Debug)]
+pub struct Credential {
+    /// Bytes of the credential. Lakers and EDHOC make no requirement on their format.
+    value: EdhocMessageBuffer,
+    /// Public authentication key (G_I or G_R) expressed in the credential
+    // For CCSs, this could be a reference into value, but Rust does not allow self-referential structs anyway.
+    public_key: BytesP256ElemLen,
```

Comment:
can you help me understand why the value and public_key are not optional, while the kid is optional?

for context, I will have to add support for an experimental PSK credential (for the emerging edhoc-psk draft), and I am trying to see how things are going to fit here. for example, I was thinking that the PSK can be stored separately (like the private keys), and the credential type might hold only the kid that identifies the PSK.